### PR TITLE
A compile fault's stack now reaches CI — #890 could never have been diagnosed

### PR DESCRIPTION
**Pull request by rbuergi**
## This does NOT fix the NRE — it makes the next occurrence diagnosable

Refs #890. **No behaviour of the compile pipeline changes.** #890 is an infrastructure fault that
aborts a compile *before Roslyn runs*, process-wide, cascading across test classes. It has been
observed **once** (run `31153582625`) and did not recur in a sweep of every non-succeeding shard job
across all attempts of the 185 CI runs since #612's diagnostics landed — 108 job logs, 0 unfetchable,
0 hits ([sweep detail](https://github.com/Systemorph/MeshWeaver/issues/890#issuecomment-5227972235)).

The problem is that **#890 as written can never be closed.** It says "the next CI occurrence will
name its throw site". It will not.

### Why the current diagnostics don't reach CI

#612 added three sinks for a non-Roslyn compile abort. Only one is observable in CI, and it is the
one **without frames**:

| Sink | Reaches CI? |
|---|---|
| `CompilationError = "{TypeName}: {Message}"` on the node | ✅ — only because a failing assertion dumps the whole `MeshNode` |
| `Compile aborted by {FullName}` + stack → **ActivityLog node** | ❌ that MeshNode is never uploaded |
| `logger.LogWarning(outcome.Error, …)` — the exception **object** | ❌ no CI artifact captures `ILogger` |

Measured, not assumed:

- The pre-#612 code **already** logged `LogWarning("Compile failure for {HubPath}: {Error}")`. In the
  attempt-1 shard log of run `31153582625` — the occurrence this issue is filed on, where **five**
  compiles failed — `Compile failure for` appears **0** times.
- A full `MeshWeaver.Hosting.Monolith.Test` `.trx` (443 KB) contains **1** `StdOut` section (a skip
  reason) and **0** `[COMPILE-TRACE]` lines, although `HandleDispatchCompile` logs one at
  `Information` on *every* compile.

The cause is in `XUnitFileLogger`: it forwards to xUnit's `ITestOutputHelper` (which the trx logger
does not persist) and to a per-method file gated behind `MESHWEAVER_TEST_FILE_LOGS` (off in CI) —
and it drops the record entirely when no test method is active, which is **exactly** the wedge case
(fixture init, teardown, and every reactive continuation on a background scheduler).

### What this PR does

**1. A fault's stack reaches the one file CI keeps.**
New `MeshWeaver.Fixture/TestTraceLog` owns the path *and the write lock* for
`%TEMP%/meshweaver-test-trace.log` — the file `dotnet-test.yml` copies into `collected-logs/`, and the
only one that survives an `exit=124` shard kill, where **no `.trx` is written at all**.
`XUnitFileLogger` now appends `Warning`+ records that carry an exception (type + full stack) there,
**before** the output-helper / `IsInTestMethod` filters that were swallowing them.

- Deliberately placed **after** `IsEnabled`: configured log levels are a production cost contract and
  are **not** widened here. No `Log*` call's level changed.
- `MonolithMeshTestBase`'s phase trace moves onto the same lock — two writers with two locks would
  interleave mid-line and corrupt both.
- Bounded at 1000 records / 2500 chars. Sized off measurement (one deliberately-failing compile test
  emits 15), and it bounds the **artifact**, not the diagnosis: the first faults in a cascade name the
  cause.

**2. A failed compile must carry a Roslyn diagnostic, not merely a non-empty string.**
`CodeEditRecompileTest` asserted only `CompilationError.Should().NotBeNullOrEmpty()` — which an
infrastructure NRE satisfies exactly as well as a real diagnostic, so a *single, non-cascading*
occurrence passed silently on a green shard. It now requires `CS####` (`FormatCompileFailure` always
writes `"{d.Id} {d.Severity}"`), turning the silent case into a red test. **This is worth more than
the artifact** — it converts a symptom we can only find by grepping logs into one CI reports itself.

**3. Delete the dead pre-#612 compile path.**
`NodeTypeCompileActivityHandler` still carried the bare `outcome.Error?.Message` that #612 replaced,
so a grep for the compile's terminal write landed on the **un-enriched dead copy first** — a triage
trap. Nothing posts `RunCompileRequest`: the cross-hub dispatch was replaced by inline `RunCompile`
(a just-created `_Activity` hub is not yet routable, so the request was dropped and the NodeType sat
at `Compiling` forever). Handler, request/response pair, handler wiring and type registrations are all
removed.

### Verification

- Solution builds `-c Release -p:CIRun=true -warnaserror` clean, after merging current `main`.
- `CodeEditRecompileTest` **11/11**; compile + release family **53/53**; `MeshWeaver.Graph.Test`
  **850/850**; `MeshWeaver.Documentation.Test` **19/19**; new `TestTraceLogFaultTest` **2/2**.
- The sink was confirmed **live**, not just compiled: a real compile-failure run wrote
  `MeshWeaver.Graph.Configuration.CompilationException` and
  `System.Reflection.TargetInvocationException` **with frames** into the collected trace file.
- The tightened assertion passes against a genuine Roslyn rejection, confirming `CS####` is really
  present (not merely assumed from reading `FormatCompileFailure`).

### Notes

- #890 stays **open** — the NRE itself is unfixed and unreproduced.
- **#618 is a SEPARATE defect** — same *shape* (a compile puts a process-wide facility into a bad
  state; the cascade crosses unrelated test classes), different mechanism. #618 is
  enumerability-after-unload in Orleans' assembly scan: `AssemblyLoadContext.Unload()` is
  asynchronous by contract, so the assembly stays in `AppDomain.CurrentDomain.GetAssemblies()` until
  the `LoaderAllocator` is collected, and the scan takes no pin. #890 involves no Orleans assembly
  enumeration and has been seen on the **Monolith** shard. Fixing #618 should **not** be expected to
  stop #890, and neither issue should block the other.
- What's New entry included: a failed build that aborts for an infrastructure reason now names the
  fault instead of showing a bare message.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/MeshWeaver.Fixture/XUnitFileOutputHelper.cs [hunk 1]
```diff
@@ -488,7 +500,6 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
         if (!outputHelper.IsInTestMethod())
             return;
 
-        var message = formatter(state, exception);
         var logEntry = $"[{logLevel}] [{_categoryName}] {message}";
```
[the remark this answers is not in the document]
Comment (rbuergi) [reply src/MeshWeaver.Fixture/XUnitFileOutputHelper.cs:510]: Good catch — fixed in aed958223.

The formatting is now lazy, and the drop conditions are checked before it happens:

```csharp
string? message = null;

// Eager ONLY on the fault path — AppendFault needs the formatted message.
if (exception is not null && logLevel >= LogLevel.Warning)
{
    message = formatter(state, exception);
    TestTraceLog.AppendFault(_categoryName, logLevel, message, exception);
}

var outputHelper = _outputHelperFactory();
if (outputHelper == null) return;
if (!outputHelper.IsInTestMethod()) return;

message ??= formatter(state, exception);   // only once the record is known to be written
```

So a non-fault record logged outside a test method — the common, high-volume case on exactly the background schedulers this change is about — is now dropped without ever being formatted, and no record is formatted twice.

Verified: `MeshWeaver.Fixture` and `MeshWeaver.Observability.Test` build `-c Release -warnaserror` clean, `TestTraceLogFaultTest` 2/2, and the whole solution builds clean with CI's flags after merging current main.
